terminus-ui/popover/src/popover-trigger.directive.spec.ts

```diff
@@ -0,0 +1,171 @@
+        fixture.detectChanges();
+        tick();
+        // console.log('popoverDebugElement.nativeElement: ', popoverDebugElement.nativeElement.getAttribute('aria-labelledby'));
+        expect(popoverDebugElement.componentInstance.myId).toEqual('example10reference1');
+      }));
+
+      test(`should fall back to the UID if no ID is passed in`, () => {
+        setup();
+        fixture.componentInstance.myId = undefined as any;
+        fixture.detectChanges();
+
+        expect(popoverDebugElement.nativeElement.getAttribute('aria-labelledby')).toEqual(expect.stringContaining('ts-popover-'));
+      });
+    });
+
+    describe(`position`, () => {
+      let event: MouseEvent;
+      beforeEach(() => {
+        event = document.createEvent('MouseEvent');
+        event.initEvent('click', true, false);
+      });
+      test(`should set popover position based on input`, fakeAsync(() => {
+        setup();
+        fixture.componentInstance.position = 'right';
+        fixture.detectChanges();
+
+        dispatchEvent(buttonDebugElement, event);
+        fixture.detectChanges();
+        tick();
+        expect(popoverDebugElement.nativeElement.getAttribute('x-placement')).toEqual('right');
+      }));
+
+      test(`should throw UI library error if provided position is not allowed`, fakeAsync(() => {
+        setup();
+        const setPosition = () => {
+          fixture.componentInstance.position = 'foo';
+          fixture.detectChanges();
+          tick();
+        };
+        expect(setPosition).toThrowError();
+      }));
+    });
+
+    describe(`hide on blur`, () => {
+      let event: MouseEvent;
+      beforeEach(() => {
+        event = document.createEvent('MouseEvent');
+        event.initEvent('click', true, false);
+      });
+
+      test(`should hide if click outside and hideOnBlur set to true`, fakeAsync(() => {
+        setup();
+        fixture.componentInstance.popoverOnHidden = jest.fn();
+        fixture.componentInstance.popoverOnShown = jest.fn();
+        fixture.detectChanges();
+        fixture.componentInstance.hideOnBlur = true;
+        fixture.detectChanges();
+        dispatchEvent(buttonDebugElement, event);
+        fixture.detectChanges();
+        tick();
+        expect(fixture.componentInstance.popoverOnShown).toHaveBeenCalled();
+        const outsideElement = fixture.debugElement.query(By.css('.outside')).nativeElement as HTMLElement;
+        dispatchEvent(outsideElement, event);
+        fixture.detectChanges();
+        tick();
+        expect(fixture.componentInstance.popoverOnHidden).toHaveBeenCalled();
+      }));
+    });
+
+    describe(`default opened`, () => {
+      test(`should open on load if defaultOpened set to true`, fakeAsync(() => {
+        setup(testComponents.DefaultOpen);
+        fixture.componentInstance.popoverOnShown = jest.fn();
+        tick();
+        expect(popoverDebugElement.nativeElement.getAttribute('x-placement')).toEqual('right');
```

Comment:
That mock call isn't necessary. But expect is expecting correctly - default open set to true, popover opens on load.

Reply:
> But expect is expecting correctly - default open set to true, popover opens on load.

Sorry, can you clarify this a bit for me? When I read this test I read that it should verify the popover is open by default. But the actual test seems to be verifying the position rather than then existence? Shouldn't we just be seeing if some parent class/element exists in the dom?

Reply:
Ah I see the confusion... so the popover element always exists. But display is hidden by default. When popover shows up, that means option set (position etc) and that `x-placement` attribute assigned. So by checking that `x-placement`, it not only makes sure popover display shows up, but also the position is set correctly (based on input). It probably could have worded better. I might change it to check `c-popover__visible` shown up first, then that placement set correctly.

Reply:
Ahhh I see

Yeah my vote would be to check for a visible class or something.. I could just see someone removing this test cause it looks like it is duplicating a positioning test etc